components/experimental/src/units/ratio.rs

```diff
@@ -193,26 +211,194 @@ impl From<u32> for IcuRatio {
 }
 
 impl FromStr for IcuRatio {
-    type Err = IcuRatioError;
+    type Err = RatioFromStrError;
+
+    /// Converts a string representation of a ratio into an `IcuRatio`.
+    /// Supported string formats include:
+    /// - Fractional notation: "1/2" becomes "1/2".
+    /// - Decimal notation: "1.5" becomes "3/2".
+    /// - Scientific notation: "1.5E6" becomes "1500000", "1.5E-6" becomes "0.0000015".
+    /// - Scientific notation with commas: "1,500E6" becomes "1500000000". Commas are disregarded.
+    /// - Integer notation: "1" becomes "1".
+    /// - Empty string: "" becomes "0".
+    /// - Fractional notation with exponent: "1/2E5" becomes "1/200000".
+    /// - Negative numbers: "-1/2" becomes "-1/2".
+    /// - Negative numbers with decimal notation: "-1.5" becomes "-3/2".
+    /// - Negative numbers with scientific notation: "-1.5E6" becomes "-1500000".
+    /// - Negative numbers with scientific notation with commas: "-1,500E6" becomes "-1500000000".
+    /// Parsing errors are returned for:
+    /// - Division by zero: "1/0".
+    /// - Multiple slashes: "1/2/3".
+    /// - Non-numeric characters in fractions: "1/2A".
+    /// - Multiple scientific notations: "1.5E6E6".
+    /// - Multiple decimal points: "1.5.6".
+    /// - Exponent part is not an integer: "1.5E6.5".
```

Comment:
suggestion: write these as doc tests to give more confidence that they are correct, and increase test coverage

Reply:
I would still like this

Reply:
Okay, I just saw it ,  I will do it in a follow-up PR.